stock_picking_operation_quick_change/wizards/stock_picking_wizard.py

```diff
@@ -94,10 +111,17 @@ def action_apply(self):
             # Write all operations destination location
             move_lines.write(vals)
         else:
+            matched_op = move_lines
             # Only write operations destination location if the location is
             # the same that old location value
-            matched_op = move_lines.filtered(
-                lambda x: x.location_dest_id == self.old_location_dest_id
-            )
+            if self.old_location_dest_id:
+                matched_op = matched_op.filtered(
+                    lambda x: x.location_dest_id == self.old_location_dest_id
+                )
+            # Only write operations destination location if the product is the same
+            if self.product_id:
+                matched_op = matched_op.filtered(
+                    lambda x: x.product_id == self.product_id
+                )
```

Comment:
¿Could you combine these two if-s in one?

Reply:
Each `if` has its own `filterered`, so they can't be combined.

Reply:
:+1: